tests/api/v2/datadog/api_logs_archives_test.go

```diff
@@ -0,0 +1,160 @@
+package test
+
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import (
+	"context"
+	"fmt"
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/DataDog/datadog-api-client-go/tests"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/h2non/gock.v1"
+	"reflect"
+	"testing"
+)
+
+// This test uses mocking because: 1) it relies on private data. 2) It relies on external services
+
+func TestLogsArchivesCreateS3(t *testing.T) {
+	archiveType := "s3"
+	action := "create"
+	testLogArchivesCreate(t, archiveType, action)
+}
+
+func TestLogsArchivesCreateAzure(t *testing.T) {
+	archiveType := "azure"
+	action := "create"
+	testLogArchivesCreate(t, archiveType, action)
+}
+
+func TestLogsArchivesCreateGCS(t *testing.T) {
+	archiveType := "gcs"
+	action := "create"
+	testLogArchivesCreate(t, archiveType, action)
+}
+
+func testLogArchivesCreate(t *testing.T, archiveType string, action string) {
+	if tests.GetRecording() == tests.ModeReplaying {
+		t.Skip("This test case does not support reply from recording")
+	}
+	ctx, finish := WithClient(WithFakeAuth(context.Background()), t)
+	client := Client(ctx)
+	assert := tests.Assert(ctx, t)
+	defer finish()
```

Comment:
move it after `ctx, finish := WithClient(WithFakeAuth(context.Background()), t)`